src/components/Steps/index.tsx

```diff
@@ -1,72 +1,181 @@
+		color: ${!ordered || completed || active
+			? '#fff'
+			: theme.colors.tertiary.semilight};
+		background-color: ${bgColor};
+		height: ${size};
+		width: ${size};
+	`;
 };
 
-const StatusIconPlaceholder = styled.span<{ status: StepProps['status'] }>`
-	font-size: ${props => px(props.theme.fontSizes[4])};
-	line-height: ${props => px(props.theme.fontSizes[4])};
-	color: ${props => getStatusColor(props.theme, props.status)};
+const StepIconBg = styled(Flex)<{
+	ordered: boolean;
+	status: statusOptions;
+	active: boolean;
+}>`
+	${props => getIconBg(props.theme, props.ordered, props.status, props.active)};
 `;
 
-const HeaderContainer = styled(Flex)`
-	min-height: ${props => px(props.theme.fontSizes[5])};
+const StepTxt = styled(Txt)<{ active: boolean }>`
+	color: ${props => (props.active ? '' : props.theme.colors.tertiary.light)};
 `;
 
-export const Step = ({ status, children, onClick }: StepProps) => {
+const StepLink = styled(Link)`
+	&:hover {
+		color: ${props => props.theme.colors.primary.main};
+	}
+`;
+
+interface StepIconProps {
+	ordered: boolean;
+	index: number;
+	active: boolean;
+	status: statusOptions;
+}
+
+const StepIcon = ({ ordered, status, active, index }: StepIconProps) => {
+	if (status === 'none') {
+		return null;
+	}
+	return (
+		<StepIconBg
+			mr={2}
+			alignItems="center"
+			justifyContent="center"
+			ordered={ordered}
+			status={status}
+			active={active}
+		>
+			{!ordered || (status === 'completed' && !active) ? (
+				<FontAwesomeIcon icon={faCheck} />
+			) : (
+				<Txt>{index + 1}</Txt>
+			)}
+		</StepIconBg>
+	);
+};
+
+export const Step = ({
+	ordered,
+	active,
+	index,
+	status,
+	children,
+	onClick,
+}: InternalStepProps) => {
 	if (typeof children !== 'string') {
 		throw new Error('The child of a Step has to be a string.');
 	}
 
 	return (
 		<Flex flexDirection="row" justifyContent="center" alignItems="center">
-			{status !== 'none' && (
-				<StatusIconPlaceholder status={status}>
-					<FontAwesomeIcon icon={faCheckCircle} />
-				</StatusIconPlaceholder>
-			)}
+			<StepIcon
+				ordered={Boolean(ordered)}
+				status={status}
+				active={Boolean(active)}
+				index={index}
+			/>
 			{onClick ? (
-				<Link ml={status === 'none' ? 0 : 2} onClick={onClick}>
+				<StepLink color="tertiary.light" onClick={onClick}>
 					{children}
-				</Link>
+				</StepLink>
 			) : (
-				<Txt ml={2}>{children}</Txt>
+				<StepTxt bold={active} active={Boolean(active)}>
+					{children}
+				</StepTxt>
 			)}
 		</Flex>
 	);
 };
 
-const FramelessSteps = ({ children, ...otherProps }: any) => {
+// A step should be clickable if:
+// - steps are not orderable OR
+// - it's located before the active step OR
+// - it's not the current step AND all previous steps are complete
+const isStepClickable = (
+	ordered: boolean,
+	steps: any[],
+	index: number,
+	activeStepIndex: number,
+) => {
+	if (!ordered) {
+		return true;
+	}
+	const allPreviousStepsComplete = every(steps.slice(0, index), {
```

Comment:
We can use `findIndex` where status is not completed, and compare that with the component index instead, so we don't have to recreate an array twice (here and below) (maybe the lodash method, the native one doesn't have the best support yet).